test/extended/setup_compat.sh

```diff
@@ -0,0 +1,354 @@
+    --node-dir="${NODE_CONFIG_DIR}" \
+    --node="${KUBELET_HOST}" \
+    --hostnames="${KUBELET_HOST}" \
+    --master="${MASTER_ADDR}" \
+    --node-client-certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" \
+    --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" \
+    --signer-cert="${MASTER_CONFIG_DIR}/ca.crt" \
+    --signer-key="${MASTER_CONFIG_DIR}/ca.key" \
+    --signer-serial="${MASTER_CONFIG_DIR}/ca.serial.txt"
+
+  ${sudo} docker run --rm \
+    --privileged --net=host -v ${MASTER_CONFIG_DIR}:${MASTER_CONFIG_DIR} \
+    openshift/origin:${version} admin \
+    create-bootstrap-policy-file --filename="${MASTER_CONFIG_DIR}/policy.json"
+
+  echo "[INFO] Creating OpenShift config"
+  ${sudo} docker run --rm \
+    --privileged --net=host -v ${MASTER_CONFIG_DIR}:${MASTER_CONFIG_DIR} \
+    -v ${VOLUME_DIR}:${VOLUME_DIR} -v ${ETCD_DATA_DIR}:${ETCD_DATA_DIR} \
+    openshift/origin:${version} start \
+    --write-config=${SERVER_CONFIG_DIR} \
+    --create-certs=false \
+    --dns="tcp://${API_HOST}:53" \
+    --listen="${API_SCHEME}://${API_BIND_HOST}:${API_PORT}" \
+    --master="${MASTER_ADDR}" \
+    --public-master="${API_SCHEME}://${PUBLIC_MASTER_HOST}:${API_PORT}" \
+    --hostname="${KUBELET_HOST}" \
+    --volume-dir="${VOLUME_DIR}" \
+    --etcd-dir="${ETCD_DATA_DIR}" \
+    --images="${USE_IMAGES}"
+
+
+  cp ${SERVER_CONFIG_DIR}/master/master-config.yaml ${SERVER_CONFIG_DIR}/master/master-config.orig.yaml
+  openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.orig.yaml --patch="{\"etcdConfig\": {\"address\": \"${API_HOST}:${ETCD_PORT}\"}}" | \
+  openshift ex config patch - --patch="{\"etcdConfig\": {\"servingInfo\": {\"bindAddress\": \"${API_HOST}:${ETCD_PORT}\"}}}" | \
+  openshift ex config patch - --type json --patch="[{\"op\": \"replace\", \"path\": \"/etcdClientInfo/urls\", \"value\": [\"${API_SCHEME}://${API_HOST}:${ETCD_PORT}\"]}]" | \
+  openshift ex config patch - --patch="{\"etcdConfig\": {\"peerAddress\": \"${API_HOST}:${ETCD_PEER_PORT}\"}}" | \
+  openshift ex config patch - --patch="{\"etcdConfig\": {\"peerServingInfo\": {\"bindAddress\": \"${API_HOST}:${ETCD_PEER_PORT}\"}}}" | \
+  openshift ex config patch - --patch="{\"imagePolicyConfig\": {\"maxImagesBulkImportedPerRepository\": ${MAX_IMAGES_BULK_IMPORTED_PER_REPOSITORY:-5}}}" > ${SERVER_CONFIG_DIR}/master/master-config.yaml
+
+  # Make oc use ${MASTER_CONFIG_DIR}/admin.kubeconfig, and ignore anything in the running user's $HOME dir
+  export ADMIN_KUBECONFIG="${MASTER_CONFIG_DIR}/admin.kubeconfig"
+  export CLUSTER_ADMIN_CONTEXT=$(oc config view --config=${ADMIN_KUBECONFIG} --flatten -o template --template='{{index . "current-context"}}')
+  local sudo="${USE_SUDO:+sudo}"
+  ${sudo} chmod -R a+rwX "${ADMIN_KUBECONFIG}"
+  echo "[INFO] To debug: export KUBECONFIG=$ADMIN_KUBECONFIG"
+}
+
+# If invoked with arguments, executes the test directly.
+function os::test::extended::focus {
```

Comment:
We should try to put all of our library functions into files that only define functions or variables so they're safe to `source`